packages/docusaurus/src/commands/deploy.ts

```diff
@@ -215,6 +215,16 @@ You can also set the deploymentBranch property in docusaurus.config.js .`);
     }
     shellExecLog('git add --all');
 
+    const gitUserName = process.env.GIT_USER_NAME;
+    if (gitUserName) {
+      shellExecLog(`git config user.name "${gitUserName}"`);
+    }
+
+    const gitUserEmail = process.env.GIT_USER_EMAIL;
+    if (gitUserEmail) {
+      shellExecLog(`git config user.email "${gitUserEmail}"`);
+    }
```

Comment:
not super fan of having config side-effects like this, and features without documentation